src/coreclr/vm/methodtable.cpp

```diff
@@ -5659,6 +5659,11 @@ void MethodTable::DoFullyLoad(Generics::RecursionGraph * const pVisited,  const
             }
             else
             {
+                if (HasVirtualStaticMethods() && !IsAbstract() && !IsSharedByGenericInstantiations())
```

Comment:
This deserves at least a TODO describing how to avoid doing this work for anything in System.Private.CoreLib. Maintaining this work as a pay-for-play feature requires that we are very careful making any changes in this part of the system.